src/testUtils.ts

```diff
@@ -117,33 +126,42 @@ export function getBenchmarkFunctions(doc: vscode.TextDocument, token: vscode.Ca
  */
 export function goTest(testconfig: TestConfig): Thenable<boolean> {
 	return new Promise<boolean>((resolve, reject) => {
-		outputChannel.clear();
+
+		// We only want to clear the outputChannel if there are no tests in flight.
+		// We do not want to clear it if tests are already running, as that could
+		// lose valuable output.
+		if (testProcesses.length < 1) {
+			outputChannel.clear();
+		}
+
 		if (!testconfig.background) {
+			let buildTags: string = testconfig.goConfig['buildTags'];
+			let args: Array<string> = ['test', ...testconfig.flags];
+			let testType: string = testconfig.isBenchmark ? 'Benchmarks' : 'Tests';
```

Comment:
@tylerb The above 3 variables do not really get used later. `buildTags` and `args` get re-assigned after this `if` block. Was this intentional? 

The net affect is that when benchmarks are run, 
- the `-benchmem` flag is not used and so the memory related stats wont be shown
- the `-run=^$` flag is not used which means all the tests will be run when the user just wants to run the single benchmark
- the timeout gets applied. I am guessing when benchmarks are run, they should be allowed to take as much time as they need and shouldnt be bound with the timeout.